ablateCore/flow/compressibleFlow.c

```diff
@@ -100,37 +100,37 @@ void CompressibleFlowComputeEulerFlux(PetscInt dim, PetscInt Nf, const PetscReal
     }
 }
 
-PetscErrorCode CompressibleFlow_SetupDiscretization(FlowData flowData, DM dm) {
+PetscErrorCode CompressibleFlow_SetupDiscretization(FlowData flowData, DM* dm) {
     PetscInt dim;
     PetscErrorCode ierr;
 
     PetscFunctionBeginUser;
-    ierr = DMSetFromOptions(dm);CHKERRQ(ierr);
+    ierr = DMSetFromOptions(*dm);CHKERRQ(ierr);
     const PetscInt ghostCellDepth = 1;
     {// Make sure that the flow is setup distributed
```

Comment:
I have an MR allowing all this crud to be done in options. I hope to have it in less then 2 weeks from now.